source/geh_calculated_measurements/tests/conftest.py

```diff
@@ -56,25 +58,51 @@ def clear_cache(spark: SparkSession) -> Generator[None, None, None]:
     spark.catalog.clearCache()
 
 
+def _get_spark():
+    spark_conf = SparkConf().setAll(
+        [
+            (k, v)
+            for k, v in {
+                "spark.sql.extensions": "io.delta.sql.DeltaSparkSessionExtension",
+                "spark.sql.catalog.spark_catalog": "org.apache.spark.sql.delta.catalog.DeltaCatalog",
+                "spark.sql.catalogImplementation": "hive",
+                "spark.sql.shuffle.partitions": "10",
+                "spark.databricks.delta.snapshotPartitions": "2",
+                "spark.driver.extraJavaOptions": f"-Ddelta.log.cacheSize=3 -XX:+CMSClassUnloadingEnabled -XX:+UseCompressedOops -Dderby.system.home={tempfile.mkdtemp()}",
+                "spark.ui.showConsoleProgress": "false",
+                "spark.ui.enabled": "false",
+                "spark.ui.dagGraph.retainedRootRDDs": "1",
+                "spark.ui.retainedJobs": "1",
+                "spark.ui.retainedStages": "1",
+                "spark.ui.retainedTasks": "1",
+                "spark.sql.ui.retainedExecutions": "1",
+                "spark.worker.ui.retainedExecutors": "1",
+                "spark.worker.ui.retainedDrivers": "1",
+                "spark.driver.memory": "2g",
+                "spark.executor.memory": "2g",
+                "spark.executor.cores": "1",
+                "spark.sql.warehouse.dir": tempfile.mkdtemp(),
+                "spark.sql.session.timeZone": "UTC",
+            }.items()
+        ]
+    )
+    session = configure_spark_with_delta_pip(
+        SparkSession.Builder().master("local").config(conf=spark_conf).enableHiveSupport()
+    ).getOrCreate()
+    session.sparkContext.setCheckpointDir(tempfile.mkdtemp())
+    return session
+
+
+_spark = _get_spark()
+
+
 @pytest.fixture(scope="session")
-def spark() -> Generator[SparkSession, None, None]:
+def spark(tmp_path_factory) -> Generator[SparkSession, None, None]:
```

Comment:
Either remove or pass on to _get_spark()
```suggestion
def spark() -> Generator[SparkSession, None, None]:
```